packages/dev/docs/pages/blog/creating-a-pointer-friendly-submenu-experience.mdx

```diff
@@ -0,0 +1,108 @@
+
+## The Shortcut Problem
+
+Submenus (or nested menus) enable multi-level exploration of menus, and even with a large number of options, users should be able to quickly find their desired option. A user should be able to hover over an item to see its submenu. Then, they should be able to move their pointer to an item in the newly opened submenu. While doing so, the pointer may leave the original item entirely and hover an unrelated item in its path towards the submenu, causing the submenu to close. We need a way to know when they’re moving their pointer to that submenu, so we can keep it open until they reach the submenu.
+
+```tsx snippet
+import {SubmenuAnimation} from './SubmenuAnimation';
+<SubmenuAnimation />
+```
+
+## Predicting User Intent
+
+We can predict the user’s intent by tracking: 
+
+* Pointer movement **direction**
+* Pointer movement **speed**
+
+We can do this by listening for [pointermove](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointermove_event) events and analyzing the changes in the pointer’s position (also called delta).
+
+## Valid Movements
+
+Now let’s imagine two lines: one from the pointer to the top of the submenu, and one from the pointer to the bottom of the submenu. We now essentially have a “range” of where the user might move their pointer on its path to the submenu. Any movement outside of this range should be considered invalid, and the submenu should close if the pointer moves outside of the original menu option.
+
+<SubmenuSafeArea
+  role="img"
+  aria-label="Diagram showing safe area triangle drawn between pointer and submenu" />
+
+## Arctangent
+
+We can measure the angle at which the pointer moved by using the 2-argument arctangent, or [atan2](https://en.wikipedia.org/wiki/Atan2). If we provide our pointer’s delta x and delta y values as arguments, we’ll get the angle (in radians) from the previous pointer position to the current pointer position.
+
+<SubmenuAtan2
+  role="img"
+  aria-label="Diagram showing atan2 value as angle between the X axis and some line on a cartesian plane" />
+
+Now we can use the atan2 function to measure the angles formed by three separate lines:
+
+* **Θ<sub>top</sub>** : Angle formed by the line from the previous pointer position to the **top inside corner** of the submenu
+* **Θ<sub>bottom</sub>** : Angle formed by the line from the previous pointer position to the **bottom inside corner** of the submenu
+* **Θ<sub>pointer</sub>** : Angle formed by the line from the previous pointer position to the **current pointer** position (delta)
+
+<SubmenuAtan2FromPointer
+  role="img"
+  aria-label="Diagram showing triangles formed between pointer and various points on submenu" />
+
+If the pointer’s delta angle is **between** the top and bottom angles, we know the user is moving their pointer in the direction of the submenu.
+
+<p style={{textAlign: 'center', fontSize: 'xx-large'}}>Θ<sub>top</sub> > Θ<sub>pointer</sub> > Θ<sub>bottom</sub></p>
+
+## Time
```

Comment:
I'm not sure if "Time" is the best title for this section. Maybe something like "Pointer Speed" since you mention speed being one of the ways of predicting user intent tho I'm not sold on that title either lol. Will ponder...

Reply:
I like "Pointer Speed" better. In fact, I think I'll rename "Arctangent" to "Pointer Direction" so they're both inline with the earlier section.